test/dtpools/src/dtpools_attr.c

```diff
@@ -9,13 +9,13 @@
 #include <inttypes.h>
 
 #define VALUE_FITS_IN_INT(val) ((val) <= INT_MAX)
-#define VALUE_FITS_IN_UINTPTR_T(val) ((val) <= (((uint64_t) 1 << (sizeof(uintptr_t) * 8 - 1)) - 1))
+#define VALUE_FITS_IN_INTPTR_T(val) (((val) * 2) <= (((uint64_t) 1 << (sizeof(uintptr_t) * 8 - 1)) - 1))
```

Comment:
I don't this is correct. If `val` is a signed int, `val * 2` may overflow -- in fact, `val` itself may already overflow -- and become negative, which will return `true` here but apparently "does not fit", right?

Reply:
`val` is `uint64_t`.  So, the assumption in the code is that `uint64_t` is large enough to accommodate it.  The problematic type would be `uintptr_t` which can be 32 bits.